lib/Dialect/TritonGPU/Transforms/Utility.cpp

```diff
@@ -673,12 +673,13 @@ Operation *cloneWithInferType(mlir::OpBuilder &rewriter, Operation *op,
   return newOp;
 }
 
-// Check if the convert will be a no-op in codegen.
+// Check if the convert will be performed without shared memory.
 static bool isFreeConvert(Operation *op) {
```

Comment:
Maybe change the function name too?

Reply:
Also, are you sure that "convert is performed without shared memory" is what you want to check?  Maybe you actually want to check "convert is performed by reordering registers, i.e. it's a nop"?  For example, a convert that does a warp shuffle maybe should not be "free"?  I don't know how it's used though.

Reply:
It's unknow whether warp shuffle with be considered as "free" or inexpensive, but we agree that better to consider only register->register conversion for now.